### PR TITLE
Revenants no longer make noise walking into gibs

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Turns the hardcoded sound that plays when you walk into some gibs into an Element and adds a check for flying to avoid playing the sound, fixing Revenants, Holoparasites, and other mobs that fly from making noise when "walking" into a pool of blood.

## Why It's Good For The Game

I was observing a revenant and heard them making noise from walking over blood, thought it was kinda f*cked up

## Changelog

:cl:
fix: Revenants (and other flying mobs) will not make noise when walking into pools of gibs,
/:cl:

### CODE REVIEW

File: code/datums/elements/squish_sound.dm [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	sound_to_play = sound
+	RegisterSignal(target, COMSIG_MOVABLE_CROSS, PROC_REF(on_cross))
+
+///Plays the set sound upon being entered, as long as the person walking into it can actually walk.
+/datum/element/squish_sound/proc/on_cross(atom/movable/source, atom/movable/crossed)
+	SIGNAL_HANDLER
+
+	if(!source.has_gravity())
+		return
+	if(!isliving(crossed) || HAS_TRAIT(crossed, TRAIT_MOVE_FLYING))
```
Comment (Ghommie) [inline code/datums/elements/squish_sound.dm:22]:
```suggestion
	if(!isliving(crossed) || || (crossed.movement_type & MOVETYPES_NOT_TOUCHING_GROUND) || crossed.throwing)
```

File: code/datums/elements/squish_sound.dm [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	. = ..()
+	sound_to_play = sound
+	RegisterSignal(target, COMSIG_MOVABLE_CROSS, PROC_REF(on_cross))
+
+///Plays the set sound upon being entered, as long as the person walking into it can actually walk.
+/datum/element/squish_sound/proc/on_cross(atom/movable/source, atom/movable/crossed)
+	SIGNAL_HANDLER
+
+	if(!source.has_gravity())
+		return
```
Comment (Ghommie) [inline code/datums/elements/squish_sound.dm:21]:
```suggestion
```
redundant with the previous suggestion.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/datums/elements/squish_sound.dm (hunk 1 above)
Comment (Fikou) [reply code/datums/elements/squish_sound.dm:22]: i think when something is being thrown it is set to flying already

Comment (Ghommie) [reply code/datums/elements/squish_sound.dm:22]: I think it isn't. Oranges said that'd easily enable prisoners to escape back in the day the gulag was separated from mining by a thin strip of lava back in the day. Though that's no longer the case.

**Pull request merged**